src/openrct2/world/Climate.cpp

```diff
@@ -181,7 +181,7 @@ extern "C"
         else if (gClimateCurrentWeatherEffect == 2)
         {
             // Create new thunder and lightning
-            uint32 randomNumber = util_rand();
+            uint32 randomNumber = scenario_rand();
```

Comment:
Its still driven by a prng which is at least sync so every client should see the thunder at the same time.